From: Gunnar Wrobel <p@rdus.de>
Subject: [PATCH] t/framework/HK/GW/Kolab_Format/ImprovedPreferencesHandling

Ensure the preferences is known and only call it if it is really available.

Signed-off-by: Gunnar Wrobel <p@rdus.de>

---
 horde-webmail/lib/Horde/Kolab/Format/XML.php |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/horde-webmail/lib/Horde/Kolab/Format/XML.php b/horde-webmail/lib/Horde/Kolab/Format/XML.php
index e66d49e..3123bee 100644
--- a/horde-webmail/lib/Horde/Kolab/Format/XML.php
+++ b/horde-webmail/lib/Horde/Kolab/Format/XML.php
@@ -968,6 +968,8 @@ class Horde_Kolab_Format_XML
      */
     function _loadMultipleCategories(&$object)
     {
+        global $prefs;
+
         if (empty($object['categories'])) {
             return;
         }
@@ -975,7 +977,8 @@ class Horde_Kolab_Format_XML
         // Create horde category if needed
         @include_once 'Horde/Prefs/CategoryManager.php';
         if ($this->_create_categories
-            && class_exists('Prefs_CategoryManager')) {
+            && class_exists('Prefs_CategoryManager')
+            && isset($prefs) && is_a($prefs, 'Prefs')) {
             $cManager = new Prefs_CategoryManager();
             $horde_categories = $cManager->get();
         } else {
-- 
tg: (7214163..) t/framework/HK/GW/Kolab_Format/ImprovedPreferencesHandling (depends on: t/framework/HK/GW/framework/Kolab_Format/WS)
-- 
TOPGIT patch commit log
=======================

commit 8320631427755105ed2ee15bec8859d423f88c91
Author: Gunnar Wrobel <p@rdus.de>
Date:   Fri Jan 30 22:43:10 2009 +0000

    Added patch release/HK-GW-Kolab_Format-PrefsHandling.patch from the mercurial release queue.
